core/src/styles.js

```diff
@@ -0,0 +1,354 @@
+
+export const border = responsiveStyle({
+  prop: 'border',
+  key: 'borders',
+  getter: getBorder
+})
+
+export const borderTop = responsiveStyle({
+  prop: 'borderTop',
+  key: 'borders',
+  getter: getBorder
+})
+
+export const borderRight = responsiveStyle({
+  prop: 'borderRight',
+  key: 'borders',
+  getter: getBorder
+})
+
+export const borderBottom = responsiveStyle({
+  prop: 'borderBottom',
+  key: 'borders',
+  getter: getBorder
+})
+
+export const borderLeft = responsiveStyle({
+  prop: 'borderLeft',
+  key: 'borders',
+  getter: getBorder
+})
+
+export const borders = props => ({
+  ...border(props),
+  ...borderTop(props),
+  ...borderRight(props),
+  ...borderBottom(props),
+  ...borderLeft(props)
+})
+borders.propTypes = {
+  ...border.propTypes,
+  ...borderTop.propTypes,
+  ...borderRight.propTypes,
+  ...borderBottom.propTypes,
+  ...borderLeft.propTypes,
+}
+
+export const borderColor = style({
+  prop: 'borderColor',
+  key: 'colors'
+})
+
+export const borderRadius = style({
+  prop: 'borderRadius',
+  key: 'radii',
+  numberToPx: true
+})
+
+export const boxShadow = style({
+  prop: 'boxShadow',
+  key: 'shadows'
+})
+
+// backgrounds
+export const background = style({
+  prop: 'background'
+})
+
+export const backgroundImage = style({
+  prop: 'backgroundImage',
+  alias: 'bgImage',
+  getter: n => `url(${n})`
+})
+
+export const backgroundSize = style({
+  prop: 'backgroundSize',
+  alias: 'bgSize',
+})
+
+export const backgroundPosition = style({
+  prop: 'backgroundPosition',
+  alias: 'bgPosition',
+})
+
+// position
+export const position = responsiveStyle({
+  prop: 'position'
+})
+
+export const zIndex = style({
+  prop: 'zIndex'
+})
+
+export const top = responsiveStyle({
+  prop: 'top',
+  numberToPx: true
+})
+
+export const right = responsiveStyle({
+  prop: 'right',
+  numberToPx: true
+})
+
+export const bottom = responsiveStyle({
+  prop: 'bottom',
+  numberToPx: true
+})
+
+export const left = responsiveStyle({
+  prop: 'left',
+  numberToPx: true
+})
+
+// pseudos
+export const hover = pseudoStyle({
+  prop: 'hover',
+  pseudoclass: 'hover',
+  keys: {
+    color: 'colors',
+    backgroundColor: 'colors',
+    borderColor: 'colors',
+    boxShadow: 'shadows'
+  }
+})
+
+export const focus = pseudoStyle({
+  prop: 'focus',
+  keys: {
+    color: 'colors',
+    backgroundColor: 'colors',
+    borderColor: 'colors',
+    boxShadow: 'shadows'
+  }
+})
+
+export const active = pseudoStyle({
+  prop: 'active',
+  keys: {
+    color: 'colors',
+    backgroundColor: 'colors',
+    borderColor: 'colors',
+    boxShadow: 'shadows'
+  }
+})
+
+export const disabled = pseudoStyle({
+  prop: 'disabledStyle',
+  pseudoclass: 'disabled',
+  keys: {
+    color: 'colors',
+    backgroundColor: 'colors',
+    borderColor: 'colors',
+    boxShadow: 'shadows'
+  }
+})
+
+// for backwards-compatibility
+// these will be removed in v3
+const __DEV__ = (process.env.NODE_ENV !== 'production')
```

Comment:
@jxnblk not every tool is able to inline constants, as it's used only once anyway I'd suggest to inline this check in the `getter` bellow